js/ui/date_box/ui.date_box.mask.js

```diff
@@ -197,15 +196,13 @@ const DateBoxMask = DateBoxBase.inherit({
 
         this._setActivePartValue(this._searchValue);
 
-        if(this.option('advanceCaret')) {
-            const isShortFormat = formatLength === 1;
-            const maxSearchLength = isShortFormat ? maxLimitLength : Math.min(formatLength, maxLimitLength);
-            const isLengthExceeded = this._searchValue.length === maxSearchLength;
-            const isValueOverflowed = parseInt(this._searchValue + '0') > max;
+        const isShortFormat = formatLength === 1;
```

Comment:
We can extract this logic to a separate method to reduce the `_searchNumber` method and make it clear